justfile

```diff
@@ -24,7 +24,7 @@ lock:
 
 # format code
 fmt:
-    black .
+    ruff format .
     blackdoc .
```

Comment:
It can't hurt, if it's not already on their radar!

I was actually going to test out, and likely suggest, switching to blacken-docs. blackdoc seems to be much less popular, and also less full-featured, than blacken-docs. The latter is also actively developed, while blackdoc is not.

Reply:
Yeah switching to blacken-docs seems better if it's actively maintained.

Reply:
@cpcloud I looked into this this morning.

> I wonder if we should bother them about adding support for docstring formatting 😁
It's already on their radar (see https://github.com/astral-sh/ruff/issues/7146), although probably not coming all that soon.

> Yeah switching to blacken-docs seems better if it's actively maintained.
There's no perfect answer here. [There seem to be some inconsistencies between the two tools](https://github.com/psf/black/issues/3083#issuecomment-1787246998), although I'd argue `blacken-docs` is doing a better job at applying Black formatting. That said, using `blacken-docs` would require wrapping all the code blocks in `pycon` Markdown fences. I also think it's not actually formatting all the code blocks in a file.

Reply:
> It looks like blacken-docs doesn't format code in docstrings

It does, but it needs to be in a `pycon` or `python` block in the docstring. This is not done in most cases in the Ibis codebase, although there are places like https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations/udf.py#L328-L337.

There's no real downside to fencing everything in `pycon` or `python` blocks, and using `blacken-docs` consistently (docs get rendered the same way), although one could argue you shouldn't need to fence them.

Reply:
Not really a huge fan of having to alter doctests just for formatting. `blackdoc` seems to do the job for now.

Reply:
Just in case come back to this thread, I actually found that `blacken-docs` isn't going to work for now, regardless: https://github.com/adamchainz/blacken-docs/issues/302#issuecomment-1792659025

Reply:
@cpcloud [Christmas came early this year!](https://astral.sh/blog/ruff-v0.1.8#formatting-code-snippets-in-docstrings) I'll test it out later today.

Reply:
That particular experience has happened to all of us, @deepyaman.